File: Compiler/TranslatorTests/TestProjects/18/TestIssue461.cs
using Bridge;
using Bridge.Html5;

namespace TestIssue461
{
    public class Issue461
    {
        public static void Test()
        {
            InputElement input = new InputElement();

            input.OnChange += (ev) =>
            {
                // Tests if ev.CurrentTarget.Value compiles
                Console.Log("ev.CurrentTarget.Value: " + ev.CurrentTarget.Value);

                // Tests if ev.IsMouseEvent() compiles
                Console.Log("IsMouseEvent: " + ev.IsMouseEvent());
            };

            AnchorElement anchor = new AnchorElement();

            anchor.OnClick += (ev) =>
            {
                // Tests if ev.CurrentTarget.Href compiles
                Console.Log("ev.CurrentTarget.Href: " + ev.CurrentTarget.Href);
            };

            // Test if Document.GetElementById<>() compiles
            DivElement div = Document.GetElementById<DivElement>("div1");

            // Tests if Element is still a superclass of all the element classes and the following code compiles
            Element element;

            element = new InputElement();
            element = new TextAreaElement();
        }
    }
}

File: Compiler/TranslatorTests/TestProjects/18/Bridge/reference/testIssue461.js
/* global Bridge */

(function (globals) {
    "use strict";

    Bridge.define('TestIssue461.Issue461', {
        statics: {
            test: function () {
                var input = document.createElement('input');
    
                input.onchange = Bridge.fn.combine(input.onchange, function (ev) {
                    // Tests if ev.CurrentTarget.Value compiles
                    console.log("ev.CurrentTarget.Value: " + ev.currentTarget.value);
    
                    // Tests if ev.IsMouseEvent() compiles
                    console.log("IsMouseEvent: " + Bridge.is(ev, MouseEvent));
                });
    
                var anchor = document.createElement('a');
    
                anchor.onclick = Bridge.fn.combine(anchor.onclick, function (ev) {
                    // Tests if ev.CurrentTarget.Href compiles
                    console.log("ev.CurrentTarget.Href: " + ev.currentTarget.href);
                });
    
                // Test if Document.GetElementById<>() compiles
                var div = document.getElementById("div1");
    
                // Tests if Element is still a superclass of all the element classes and the following code compiles
                var element;
    
                element = document.createElement('input');
                element = document.createElement('textarea');
            }
        }
    });
    
    
    
    Bridge.init();
})(this);
